### PR TITLE
Add io.github.limads.Queries

### PR DESCRIPTION
**Pull request by limads**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

Hello, I'd like to submit my application to Flathub. It is building successfully with `flatpak-builder` on my machine. Let me know if there are any issues.

### CODE REVIEW

File: com.github.limads.Queries.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	        
+	        {
+	            "type" : "archive",
+	            "url" : "https://limads.sfo3.cdn.digitaloceanspaces.com/queries-vendored/v0.1/vendored.tar.gz",
+	            "sha256" : "7add2c5672450ad4e48fe1d84d921524260a2c0c27c826f6dc6b6727787410d8"
+	        },
+	        
+		    {
+		        "type" : "git",
+		        "url" : "https://github.com/limads/queries.git"
```
Comment (hfiguiere) [inline com.github.limads.Queries.json:50]: you need a commit or tg here.

File: com.github.limads.Queries.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		    }
+		    
+	        ],
+	        "post-install" : [
+	        	"sh ${FLATPAK_BUILDER_BUILDDIR}/scripts/postinstall.sh"
```
Comment (hfiguiere) [inline com.github.limads.Queries.json:55]: I had a look at this script and it does many things wrong.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.github.limads.Queries.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
+{
+    "app-id" : "com.github.limads.Queries",
```
[the remark this answers is not in the document]
Comment (limads) [reply com.github.limads.Queries.json:2]: Shouldn't the app id reflect the reverse repository URL?

File: com.github.limads.Queries.json (hunk 2 above)
Comment (limads) [reply com.github.limads.Queries.json:55]: Thanks for taking a look. I've just examined the script and failed to see the error - The executable and the symbolic/application icons are all found after the installation.

File: com.github.limads.Queries.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "42",
```
[the remark this answers is not in the document]
Comment (limads) [reply com.github.limads.Queries.json:4]: Can confirm the application builds on the 43 runtime. Updated.

File: com.github.limads.Queries.json (hunk 1 above)
Comment (limads) [reply com.github.limads.Queries.json:50]: Ok, updated.

**Comment by hfiguiere** [comment]
bot, build com.github.limads.Queries

File: com.github.limads.Queries.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply com.github.limads.Queries.json:2]: hosting sites like github and gitlab have delegated domain for "pages" that are more approrpriate.

File: com.github.limads.Queries.json (hunk 2 above)
Comment (hfiguiere) [reply com.github.limads.Queries.json:55]: 1. it's all specific to flatpak becaus it hardcode a prefix.
2. it should never put anything in `share/app-info/`. Not sure what piece of documentation told you to install anything in there.
3. it needlessly use mkdir when install handle creating directories.
4. icons arent even in the right place
5. this should either be in a proper build system (cargo is too limited) or put as individual commands in the manifest here.

File: com.github.limads.Queries.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply com.github.limads.Queries.json:2]: it's in the documentation.

File: com.github.limads.Queries.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply com.github.limads.Queries.json:4]: it's not

File: com.github.limads.Queries.json [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command" : "queries",
+    "finish-args" : [
+        "--share=network",
+        "--device=dri",
+        "--socket=wayland",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--filesystem=home"
```
Comment (hfiguiere) [inline com.github.limads.Queries.json:16]: why do you need home acces?

Comment (limads) [reply com.github.limads.Queries.json:16]: The application requires access to files at paths specified by the user (SSL certificates) and there is no way to anticipate where the user is going to store them. Plus, a lot of the interactions with scripts are not based on file chooser dialogs, therefore the home access.

Comment (JakobDev) [reply com.github.limads.Queries.json]: If the App uses a Gtk Filepicker, the App will get access to the files the User picked, so you don't need home access.

Comment (limads) [reply com.github.limads.Queries.json]: There are certain places it doesn't use a file chooser dialog.

**Comment by limads** [comment]
Ok, git tag for source repository, application id and post-install commands edited on the last commit.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
also the indentation is all over the place making the manifest harder to read.

**Comment by limads** [comment]
Done, changes applied in the last commit.

**Comment by hfiguiere** [comment]
bot, build com.github.limads.Queries

**Comment by hfiguiere** [comment]
bot, build io.github.limads.Queries

**Comment by hfiguiere** [comment]
AppStream validation failed.

```
builddir/files/share/appdata/io.github.limads.Queries.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• tag-missing           : <name> is not present
Validation of files failed
```

**Comment by limads** [comment]
Just updated the appdata file to add <name> and <content_rating> tags.

**Comment by barthalion** [comment]
bot, build io.github.limads.Queries

**Comment by limads** [comment]
Invite accepted, thank you all!

**Pull request closed without merging**